Add Sphinx docs

cdeil opened the pull request:
This PR add Sphinx docs to address #12 and moves the content from `README.md` to `docs/index.rst` as discussed in #86 .

For now I've kept the single-page approach. As the documentation grows and API docs are added, it could be split into multiple pages.

I've kept the README as-is for now. Maybe we first merge this, make sure the docs are up and look OK, and then write a simple, short README in a follow-up PR?

@andycasey - Can you please review and activate `ads` on readthedocs?

cdeil commented:
Here's a screenshot of what the HTML docs look like for me locally (should look identical on readthedocs):

<img width="1058" alt="screen shot 2017-01-05 at 12 39 50" src="https://cloud.githubusercontent.com/assets/852409/21679389/1fe1d1b4-d344-11e6-9a17-f2fefb9288f9.png">

vsudilov commented:
Looks good, but I'm concerned about the duplication of info. Is there a way to use the same file(s) as a source for all outputs (github, pypi, read the docs)?

cdeil commented:
> Looks good, but I'm concerned about the duplication of info. Is there a way to use the same file(s) as a source for all outputs (github, pypi, read the docs)?

My suggestion would be to have the docs in one place, and not try to duplicate it in two or three places.

My suggestion would be to organise the info roughly like for this package:
- Docs are only on RTD: http://astropy-regions.readthedocs.io/
- README only points to repo and docs: https://github.com/astropy/regions/blob/master/README.md
- PyPI only points to repo and cos: https://pypi.python.org/pypi/regions/0.1

Like I said, my suggestion would be to remove the content of the README only in a follow-up PR, once this one is merged and docs look OK on readthedocs. I will make a follow-up PR to clean up the README and the long_description, so that the PyPI page will then look OK when you make the next release.

OK to do it this way?
(or do you prefer that I make more changes in this PR?)

cdeil commented:
> Looks good, but I'm concerned about the duplication of info. Is there a way to use the same file(s) as a source for all outputs (github, pypi, read the docs)?

> I'd prefer to keep at least a short quickstart in the README.md rendered on github, and to do everything in this PR rather than breaking it up into two (can always make another PR if something is incorrect after the fact).

@vsudilov - I've added a commit (2c76550) here removing the duplicated docs content from the README. I've also changed the format from MD to RST, so that it will render correctly on PyPI on the next release.

> I'd prefer to keep at least a short quickstart in the README.md rendered on github

I don't know which content you'd like to keep in the README. Like I said I wouldn't duplicate any content.

Can you please do it yourself? (as a maintainer you can add commits to this branch)

jonnybazookatone commented:
I think he's just suggesting keeping the pip install and a simple example of how to do a query. Maybe something as simple as:

**A Python Module to Interact with NASA's ADS that Doesn't Suck™**
==================================================================

[![Build Status](http://img.shields.io/travis/andycasey/ads.svg?branch-master)](https://travis-ci.org/andycasey/ads) [![Coverage Status](https://coveralls.io/repos/github/andycasey/ads/badge.svg?branch=master)](https://coveralls.io/github/andycasey/ads?branch=master) [![PyPi download count image](http://img.shields.io/pypi/dm/ads.svg)](https://pypi.python.org/pypi/ads/)

If you're in astro research, then you pretty much _need_ NASA's ADS. It's tried, true, and people go crazy on the rare occasions when it goes down.

**Getting Started**

1. You'll need an API key from NASA ADS labs. Sign up for the newest version of ADS search at https://ui.adsabs.harvard.edu, visit account settings and generate a new API token. The official documentation is available at https://github.com/adsabs/adsabs-dev-api

2. When you get your API key, save it to a file called ``~/.ads/dev_key`` or save it as an environment variable named ``ADS_DEV_KEY``

3. From a terminal type ``pip install ads`` (or [if you must](https://stackoverflow.com/questions/3220404/why-use-pip-over-easy-install), use ``easy_install ads``)

4. You can use this module to search for some popular supernova papers:
````python
>>> import ads

# Opps, I forgot to follow step 2 in "Getting Started"
>>> ads.config.token = 'my token'

>>> papers = ads.SearchQuery(q="supernova", sort="citation_count")

>>> for paper in papers:
>>>    print(paper.title)
   ...:
[u'Maps of Dust Infrared Emission for Use in Estimation of Reddening and Cosmic Microwave Background Radiation Foregrounds']
[u'Measurements of Omega and Lambda from 42 High-Redshift Supernovae']
[u'Observational Evidence from Supernovae for an Accelerating Universe and a Cosmological Constant']
[u'First-Year Wilkinson Microwave Anisotropy Probe (WMAP) Observations: Determination of Cosmological Parameters']
[u'Abundances of the elements: Meteoritic and solar']
````

For further details on how to use `ads` see:
 * Documentation: https://ads.readthedocs.io/
 * GitHub repository: https://github.com/andycasey/ads
 * PyPI: https://pypi.python.org/pypi/ads

vsudilov commented:
Yeah, that's basically the idea -- though I'd make it even shorter personally.

The README.rst needs syntax updating for badges to render I think.

I can take over if you prefer; @andycasey is the reviewer so happy to go with whatever he prefers.

cdeil commented:
> I can take over if you prefer

Thanks!

vsudilov commented:
@andycasey , I think the next step is to sign up for readthedocs.org. I'd suggest you go ahead and own the account and create an org (http://docs.readthedocs.io/en/latest/business/organizations.html) so that contributors can push to it.

I'm happy to own it if you'd rather -- let me know.

jonnybazookatone commented:
If you didn't notice you'll need to fix the merge conflict. Most likely the addition made in #90 was not in the removed README.md of this PR.

jonnybazookatone commented:
@andycasey have you gotten around to setting up a readthedocs account?

andycasey commented:
Thanks for pinging me on this @jonnybazookatone! I did set up an account, but it wasn't building until this was merged. It's online now on RTD. Thanks @cdeil, @vsudilov, and @jonnybazookatone!

jonnybazookatone commented:
Cool, looks great!

@aaccomazzi: FYI, if you want to link directly to the docs at any point: http://ads.readthedocs.io/en/latest/

cdeil commented:
Thank you!

aaccomazzi commented:
Thanks, updated our documentation to point to it (https://github.com/adsabs/adsabs-dev-api/blob/master/README.md)
